src/options.cc

```diff
@@ -2,22 +2,27 @@
 
 #include "options.h"
 
-Options::Options(const char* file_name) {
+void Options::loadOptionsFromFile(const char* filename) {
```

Comment:
How about we make this return a `bool`, and if we successfully parsed the config file and loaded everything OK,
we return `true`, and if there was any sort of problem, we return `false`?
See my comment below too